workerEngineReduce.py

```diff
@@ -350,59 +279,79 @@ def pick_candidate(self): #how the next best move is picked
-
-                    
-
+                                }
 
         logging.debug(f'best move is - {best_move} & win rate is - {potency} & lower bound win rate is - {lb_potency}')
         return moves, best_move, potency, (lb_potency, ub_potency), n
 
     def find_opponent_move(self, move):
-        try: # find the odds of the pgn opponent move in the opening stats from the API.
-            if move.uci() == 'e8g8': #Change the values for castling in Universal Chess Interface codes, can ignore
-                move_uci = 'e8h8'
-            elif move.uci() == 'e1g1':
-                move_uci = 'e1h1'
-            else:
-                move_uci = move.uci()
+        if move.uci() == 'e8g8': #Change the values for castling in Universal Chess Interface codes, can ignore
+            move_uci = 'e8h8'
+        elif move.uci() == 'e1g1':
+            move_uci = 'e1h1'
+        else:
+            move_uci = move.uci()
 
+        try: # find the odds of the pgn opponent move in the opening stats from the API.
             move_stats = next(item for item in self.stats['moves'] if item["uci"] == move_uci) #move stats is the next move in self stats moves which matches the move fed into function
         except:
-            raise Exception(f'Cannot find move {move_uci} in opening explorer API response') 
+            self.status.error(f"Failed to find move {move_uci} in the Opening Explorer API response", f"FEN: {self.fen}")
+            raise Exception(f'Cannot find move {move_uci} in opening explorer API response')
 
         chance = move_stats['total_games'] / self.stats['total_games'] #total games for next move overf total games for current move
         #print("move opponent played =",move, " & percent chance of them playing it =", chance) #prints move opponent played from each position
         #print (move_stats)
         return move_stats, chance
 
-
-    def create_children(self):
-
-        children = []
-        for i, m in enumerate(self.stats['moves']):
-            worker = self.play(m['san'])
-            children.append(worker)
-        return children
-    
-    
-
-
     def find_move_tree(self): #how we return possible opponent continuations
-
         return self.stats['moves']
 
     def find_potency(self): #how we return possible opponent continuations
         stats = self.stats
-        stats['white_perc'], stats['black_perc'], stats['draw_perc'], stats['total_games'] = calc_percs(stats['white'], stats['black'], stats['draws'])        
+        stats['white_perc'], stats['black_perc'], stats['draw_perc'], stats['total_games'] = self.calc_percs(stats['white'], stats['black'], stats['draws'])
         if self.board.turn == chess.WHITE: #if we are white
             potency = stats['black_perc']
             draws = stats['draw_perc']
         else:
             potency = stats['white_perc']
             draws = stats['draw_perc']
         return potency, stats ['total_games'], draws
-    
 
-def quitEngine():

```

Comment:
There is now only one engine object, passed from `BookBuilder.py`, so there's no reason to have this method here